test/token/ERC20/ERC20Channel.test.js

```diff
@@ -0,0 +1,126 @@
+import { increaseTimeTo, duration } from '../../helpers/increaseTime';
+import EVMRevert from '../../helpers/EVMRevert';
+var ethUtils = require('ethereumjs-util');
+
+var BigNumber = web3.BigNumber;
+
+require('chai')
+  .use(require('chai-as-promised'))
+  .use(require('chai-bignumber')(BigNumber))
+  .should();
+
+var StandardTokenMock = artifacts.require('StandardTokenMock.sol');
+var ERC20Channel = artifacts.require('ERC20Channel.sol');
+var ECRecovery = artifacts.require('ECRecovery.sol');
+
+contract('ERC20Channel', function ([_, receiver, sender]) {
+  var token;
+  var tokenChannel;
+
+  // This private keys should match the ones used by testrpc
+  const privateKeys = {};
+  privateKeys[receiver] = '0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501201';
+  privateKeys[sender] = '0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501202';
+
+  // Sign a message with a private key, it returns the signature in rpc format
+  function signMsg (msg, pvKey) {
+    const sig = ethUtils.ecsign(ethUtils.toBuffer(msg), ethUtils.toBuffer(pvKey));
+    return ethUtils.toRpcSig(sig.v, sig.r, sig.s);
+  }
+
+  beforeEach(async function () {
+    const ecrecovery = await ECRecovery.new();
+    ERC20Channel.link('ECRecovery', ecrecovery.address);
+    token = await StandardTokenMock.new(sender, 100);
+  });
+
+  it('create channel', async function () {
+    tokenChannel = await ERC20Channel.new(token.address, receiver, duration.days(1), { from: sender });
+    await token.transfer(tokenChannel.address, 60, { from: sender });
+    const channelInfo = await tokenChannel.getInfo();
+    assert.equal(parseInt(channelInfo[0]), 60);
+    assert.equal(parseInt(channelInfo[1]), 0);
+    assert.equal(parseInt(channelInfo[2]), 0);
+  });
+
+  it('create channel and close it with a mutual agreement from sender', async function () {
+    tokenChannel = await ERC20Channel.new(token.address, receiver, duration.days(1), { from: sender });
+    await token.transfer(tokenChannel.address, 30, { from: sender });
+    const hash = await tokenChannel.generateBalanceHash(20);
+    const senderSig = signMsg(hash, privateKeys[sender]);
+    assert.equal(sender,
+      await tokenChannel.getSignerOfBalanceHash(20, senderSig)
+    );
+    const senderHash = await tokenChannel.generateKeccak256(senderSig);
+    const closingSig = signMsg(senderHash, privateKeys[receiver]);
+    const channelInfo = await tokenChannel.getInfo();
+    assert.equal(parseInt(channelInfo[0]), 30);
+    assert.equal(parseInt(channelInfo[1]), 0);
+    assert.equal(parseInt(channelInfo[2]), 0);
+    await tokenChannel.cooperativeClose(20, senderSig, closingSig, { from: receiver });
+    (await token.balanceOf(sender)).should.be.bignumber
+      .equal(80);
+    (await token.balanceOf(receiver)).should.be.bignumber
+      .equal(20);
+  });
+
+  it('create channel and close it with a mutual agreement from receiver', async function () {
```

Comment:
WDTY about dynamically generating this test and the previous one using a loop over `[sender, receiver, anotherAddress]`, in order not to duplicate code?

Reply:
to select three random addresses from the 10 ones being used by testrpc? If it is that what you mean I think it will be better.

Reply:
Sorry for the confusion: I meant to dynamically generate the tests `create channel and close it with a mutual agreement from sender` and `create channel and close it with a mutual agreement from receiver` since they share most of the code. So, instead of having two tests that duplicate most of the code, do something like:

```js
[['sender', sender], ['receiver', receiver]].forEach(function(description, who) {
   it(`create channel and close it with a mutual agreement from ${description}`, async function () {
     tokenChannel = await ERC20Channel.new(token.address, receiver, duration.days(1), { from: sender });
     await token.transfer(tokenChannel.address, 30, { from: sender });
     const hash = await tokenChannel.generateBalanceHash(20);
     const senderSig = signMsg(hash, senderPrivateKey);
     assert.equal(sender,
       await tokenChannel.getSignerOfBalanceHash(20, senderSig)
     );
     const senderHash = await tokenChannel.generateKeccak256(senderSig);
     const closingSig = signMsg(senderHash, receiverPrivateKey);
     const channelInfo = await tokenChannel.getInfo();
     assert.equal(parseInt(channelInfo[0]), 30);
     assert.equal(parseInt(channelInfo[1]), 0);
     assert.equal(parseInt(channelInfo[2]), 0);
     await tokenChannel.cooperativeClose(20, senderSig, closingSig, { from: who });
     (await token.balanceOf(sender)).should.be.bignumber
       .equal(80);
     (await token.balanceOf(receiver)).should.be.bignumber
       .equal(20);
   });
});
```

Or define a helper function that runs the test:

```js
function createAndCloseChannelWithMutualAgreementFrom(description, who) {
  it(`create channel and close it with a mutual agreement from ${description}`, async function () {
    tokenChannel = await ERC20Channel.new(token.address, receiver, duration.days(1), { from: sender });
    await token.transfer(tokenChannel.address, 30, { from: sender });
    const hash = await tokenChannel.generateBalanceHash(20);
    const senderSig = signMsg(hash, senderPrivateKey);
    assert.equal(sender,
      await tokenChannel.getSignerOfBalanceHash(20, senderSig)
    );
    const senderHash = await tokenChannel.generateKeccak256(senderSig);
    const closingSig = signMsg(senderHash, receiverPrivateKey);
    const channelInfo = await tokenChannel.getInfo();
    assert.equal(parseInt(channelInfo[0]), 30);
    assert.equal(parseInt(channelInfo[1]), 0);
    assert.equal(parseInt(channelInfo[2]), 0);
    await tokenChannel.cooperativeClose(20, senderSig, closingSig, { from: who });
    (await token.balanceOf(sender)).should.be.bignumber
      .equal(80);
    (await token.balanceOf(receiver)).should.be.bignumber
      .equal(20);
  });
};

createAndCloseChannelWithMutualAgreementFrom('receiver', receiver);
createAndCloseChannelWithMutualAgreementFrom('sender', sender);
createAndCloseChannelWithMutualAgreementFrom('other', other);
```

That said, it's also interesting to have a random addresses helper. I'd add an option to seed it with a particular value though, so tests are reproducible.